libsplinter/src/admin/store/diesel/models.rs

```diff
@@ -0,0 +1,540 @@
+            authorization_type: String::from(&proposed_circuit.authorization_type),
+            persistence: String::from(&proposed_circuit.persistence),
+            durability: String::from(&proposed_circuit.durability),
+            routes: String::from(&proposed_circuit.routes),
+            circuit_management_type: proposed_circuit.circuit_management_type.clone(),
+            application_metadata: proposed_circuit.application_metadata.clone(),
+            comments: proposed_circuit.comments.clone(),
+        }
+    }
+}
+
+/// Database model representation of a `VoteRecord`
+#[derive(Debug, PartialEq, Associations, Identifiable, Insertable, Queryable, QueryableByName)]
+#[table_name = "vote_record"]
+#[belongs_to(CircuitProposalModel, foreign_key = "circuit_id")]
+#[primary_key(circuit_id, voter_node_id)]
+pub struct VoteRecordModel {
+    pub circuit_id: String,
+    pub public_key: Vec<u8>,
+    pub vote: String,
+    pub voter_node_id: String,
+}
+
+impl From<&CircuitProposal> for Vec<VoteRecordModel> {
+    fn from(proposal: &CircuitProposal) -> Self {
+        proposal
+            .votes
+            .iter()
+            .map(|vote| VoteRecordModel {
+                circuit_id: proposal.circuit_id.clone(),
+                public_key: vote.public_key.clone(),
+                vote: String::from(&vote.vote),
+                voter_node_id: vote.voter_node_id.clone(),
+            })
+            .collect()
+    }
+}
+
+impl TryFrom<&VoteRecordModel> for VoteRecord {
+    type Error = AdminServiceStoreError;
+    fn try_from(vote: &VoteRecordModel) -> Result<Self, Self::Error> {
+        Ok(VoteRecord {
+            public_key: vote.public_key.clone(),
+            vote: Vote::try_from(vote.vote.clone())?,
+            voter_node_id: vote.voter_node_id.clone(),
+        })
+    }
+}
+
+/// Database model representation of a `ProposedNode`
+#[derive(Debug, PartialEq, Associations, Identifiable, Insertable, Queryable, QueryableByName)]
+#[table_name = "proposed_node"]
+#[belongs_to(ProposedCircuitModel, foreign_key = "circuit_id")]
+#[primary_key(circuit_id, node_id)]
+pub struct ProposedNodeModel {
+    pub circuit_id: String,
+    pub node_id: String,
+}
+
+impl From<&ProposedCircuit> for Vec<ProposedNodeModel> {
+    fn from(proposed_circuit: &ProposedCircuit) -> Self {
+        proposed_circuit
+            .members
+            .iter()
+            .map(|node| ProposedNodeModel {
+                circuit_id: proposed_circuit.circuit_id.clone(),
+                node_id: node.node_id.clone(),
+            })
+            .collect()
+    }
+}
+
+/// Database model representation of the endpoint values associated with a `ProposedNode`
+#[derive(Debug, PartialEq, Associations, Identifiable, Insertable, Queryable, QueryableByName)]
+#[table_name = "node_endpoint"]
+#[belongs_to(ProposedNodeModel, foreign_key = "node_id")]
+#[primary_key(node_id, endpoint)]
+pub struct NodeEndpointModel {
```

Comment:
We should still have ProposdeNodeEndpointModel for endpoint that are a part of a proposal. The NodeEndpointModel of approved circuits should not be stored in the same table.

Reply:
This is because the Proposals should include the full circuit (including full member definitions). If someone were to go in and look at this table directly, they could not tell the difference between an approved member/node and only a proposed one.

Reply:
I added back a `proposed_node_endpoint` table (and moved the `node_endpoint` so the organization makes sense) and also renamed the `circuit_member` field from `member` -> `node_id` to make the association with `node_endpoint` more clear